src/os/ui/file/addserver.js

```diff
@@ -0,0 +1,176 @@
+goog.module('os.ui.file.AddServer');
+goog.module.declareLegacyNamespace();
+
+const FileParserConfig = goog.require('os.parse.FileParserConfig');
+const Module = goog.require('os.ui.Module');
+const ImportManager = goog.require('os.ui.im.ImportManager');
+const {ROOT} = goog.require('os');
+const osWindow = goog.require('os.ui.window');
+const WindowEventType = goog.require('os.ui.WindowEventType');
+
+
+/**
+ * The addserver directive.
+ * @return {angular.Directive}
+ */
+const directive = () => {
+  return {
+    restrict: 'E',
+    replace: true,
+    scope: true,
+    templateUrl: ROOT + 'views/file/addserver.html',
+    controller: Controller,
+    controllerAs: 'ctrl'
+  };
+};
+
+
+/**
+ * The element tag for the directive.
+ * @type {string}
+ */
+const directiveTag = 'addserver';
+
+
+/**
+ * Add the directive to the module.
+ */
+Module.directive(directiveTag, [directive]);
+
+
+/**
+ * Controller for the addserver directive.
+ * @unrestricted
+ */
+class Controller {
+  /**
+   * Constructor.
+   * @param {!angular.Scope} $scope
+   * @param {!angular.JQLite} $element
+   * @ngInject
+   */
+  constructor($scope, $element) {
+    /**
+     * @type {?angular.Scope}
+     * @private
+     */
+    this.scope_ = $scope;
+
+    /**
+     * @type {?angular.JQLite}
+     * @private
+     */
+    this.element_ = $element;
+
+    /**
+     * @type {boolean}
+     */
+    this.loading = false;
+
+    /**
+     * @type {string}
+     */
+    this.serverType = '';
```

Comment:
Sorry one more thing. The `serverType` and `items` properties are both referenced by the template, and as such should use bracket notation to avoid being renamed. It looks like the compiler isn't renaming them because of some unrelated externs, but that shouldn't be relied on because the behavior could change.